### ISSUE TITLE
Provide easy tool to compile DSL to C# and include compiled jar

### ISSUE THREAD

**Issue by Misiu** [ngs-doo/dsl-compiler-client#75]
I just downloaded .NET library and in one of folders I found DSL files, below is one of them:

```
module ReportCommon
{
  value BasicInfo {
    ScrapeReport     scrapeReport;
    Bank             bank;
    String           source;
    String?          username;
    String?          address;
    String?          phone;
    String?          email;
    List<MiscEntry>  miscEntryList;
  }

  value ScrapeReport {
    Date     scrapeDate;    // scrape date
    String   status;        // status of scrape "ok", "invalid-login", "scrape-error" etc.
    String   comment;       // status comment "Process finished successfuly", etc.
    Int      reportNumber;  // ordinal number of report
    Boolean  isFinal;       // if report final
  }

  value Bank {
    String  country;         // ISO-3316: "SE", "ES", "FI"
    String  abbreviation;    // "op.fi"
    String  name;            // "Ossuspankki"
  }

  // TODO
  value MiscEntry {
    String  provider;   // trustbuddy.se
    String  key;        // user_id
    String  value;      // 12341234
  }

  value Scrape {
    ScrapeReport    scrapeReport;
    Bank            bank;
    UserDetails?    userDetails;    // it can be null in case of corporate account, in which cas companyDetails will be present
    CompanyDetails? companyDetails; // see above

    List<Account>   accountList;    // list of bank accounts
  }

  value UserDetails {
    String  name;
    String? address;
    String? phone;
    String? email;
  }

  value CompanyDetails {
    String  name;
    String? organizationNumber;
    String? address;
    String? phone;
    String? email;
  }

  value Account {
    String  number;
    String  kind;     // giro, currency
    Money   balance;  //
    String  currency; // infer "EUR"
    String? iban;
    String? holderName; // user can have access to bank accounts from different person in which case holderName will be different from UserDetails.name

    List<Transaction>  transactionList;
  }

  value Transaction {
    Date    onDate;
    String  description;
    Money   amount;    // if negative, then outgoing
    Money   balance;   // status of the account after this transaction
    Map     params;    // assume of no interset
  }
}

```

I was searching a bit and found this repository, but for someone without JAVA experience and without JAVA tools on his PC it will be hard to compile source code.

If it will be possible please add build output to each release, these way people like me would be able to compile DSL files to C# source.

If there is online tool that can do such things please let me know.
Thanks!

**Comment by zapov** [comment]
There is Visual studio plugin which you can use: https://marketplace.visualstudio.com/items?itemName=RikardPavelic.DDDforDSLPlatform

It doesn't require Java and will create DLLs for you. I kind of avoid exposing C# source to discourage using it directly... but I guess I can bring back it there if there are good reasons for it.

**Comment by Misiu** [comment]
@zapov as I wrote I got DSL files with library I want to use. All I need are classes that are generated by compiling DSL. I've decompiled DLL and got classes from there, but if it is possible I'd like to get classes from DSL. I need just models, without all database related methods and properties.

**Comment by zapov** [comment]
Yes. Depending on what you target you will get different stuff. Revenj will give you DB stuff. POCO will give you classes only

**Comment by Misiu** [comment]
@zapov POCO is exactly what I need 😄

**Comment by zapov** [comment]
I'm not sure if that's a question, but with VS plugin you can target POCO.

**Comment by Misiu** [comment]
@zapov  But will I see source of POCO's or will I get only DLL?

**Comment by zapov** [comment]
You will only get the DLL (sources will be kept in memory only)
Currently the easiest way to get the source code is to use the Java tool.
It's not that complicated. This should work 

    java -jar dsl-clc.jar target=dotnet_poco dsl=model.dsl source-only

You can get source in JSON/XML format directly from the compiler... but you will still have to unpack it into separate files.

**Comment by Misiu** [comment]
@zapov can I download `dsl-clc.jar` already compiled? I have JAVA runtime, but nothing else.

**Comment by zapov** [comment]
Yes. I see it's missing from recent releases... but older ones should work too: https://github.com/ngs-doo/dsl-compiler-client/releases/tag/1.8.4

**Comment by Misiu** [comment]
@zapov Thank You for link. I've downloaded `dsl-clc.jar`. I've tried compiling DSL files using command `java -jar dsl-clc.jar target=dotnet_poco dsl=Dsl source-only`, where `Dsl` if folder in which I have my dsl files.
I get no errors in console, but nothing is generated. when I remove `source-only` I get DLL file and empty `dotnet_poco` folder.
When I call: `java -jar dsl-clc.jar help=source-only` I get `Unknown command: source-only`.

Sorry for all the trouble.
